commandapi-platforms/commandapi-bukkit/commandapi-bukkit-vh/src/main/java/dev/jorel/commandapi/CommandAPIVersionHandler.java

```diff
@@ -49,11 +49,13 @@ public interface CommandAPIVersionHandler {
 	 * @return an instance of NMS which can run on the specified Minecraft version
 	 */
 	static CommandAPIPlatform<?, ?, ?> getPlatform() {
+		CommandAPIPlatform<?, ?, ?> latestNMS = new NMS_1_21_R1();
 		if (CommandAPI.getConfiguration().shouldUseLatestNMSVersion()) {
-			return new NMS_1_21_R1();
+			return latestNMS;
```

Comment:
This applies to `shouldUseLatestNMSVersion` and `shouldBeMoreLenientForMinorVersions`, but maybe there could be a warning message logged to make it clear when we're loading a potentially incompatible NMS version.